### DISCUSSION TITLE
Some problem

### DISCUSSION THREAD

**Discussion by m5it** [Dlloydev/ESP32-ESP32S2-AnalogWrite#19] [General]
pwmWrite.cpp:211:7: error: 'ledc_channel_config_t' has no non-static data member named 'flags'

any idea what to do?

**Comment by Dlloydev** [comment]
Try the latest versions: 
Arduino ESP32 Version 2.0.5
ESP32-ESP32S2-AnalogWrite 3.0.4

**Reply by m5it** [reply]
thanks for answer. will test. noticing you. thanks! didnt test yet but think this will work. probably because of update of ide this thing should be updated too. not sure. anyway will respond soon.*

**Reply by m5it** [reply]
man.! now is working back. a lot of thank you**** and to all who managed to work back*** love you*** :)

**Comment by evup4s** [comment]
I noticed, when trying out the pwmWrite as a 'WokWi demo', and then viewing the logged signal file with 'PulseView' software, that this indicates a frequency (on all 3 channels) of 200Hz instead of the 'planned/programmed' 100Hz. Is this an error of me, of PulseView, or just a matter of insufficient setup of frequency divider in the ESP32 setup part?
![3phase wokwi signals PWM 100Hz or 200Hz](https://user-images.githubusercontent.com/93977722/197342261-276836ad-424e-49d4-b619-f27dfe3c0d93.PNG)

**Comment by Dlloydev** [comment]
Thanks for finding this. Looks like this could be a recent WokWi ESP32S2 simulator issue (will let them know).

The example `ESP32_S2_pwmWrite.ino` tested OK on my hardware:

ESP32-ESP32S2-AnalogWrite 3.0.4
Arduino 1.8.19
ESP32 2.0.5
Board: "ESP32S2 Dev Module"
Analyzer: Saleae Logic Pro 8 using Saleae Logic 1.2.18

![image](https://user-images.githubusercontent.com/63488701/197346756-e355905b-839d-493e-aafb-d5475d680719.png)

**Reply by Dlloydev** [reply]
They've fixed the issue (tested using their ESP32-S2 simulation board).
